crates/uv-configuration/src/dev.rs

```diff
@@ -1,445 +1,350 @@
+}
+
+impl std::ops::Deref for DevGroupsSpecification {
+    type Target = DevGroupsSpecificationInner;
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
 
+impl DevGroupsSpecificationInner {
     /// Returns `true` if the specification allows for production dependencies.
+    ///
+    /// (This is really just asking if an --only flag was passed.)  
     pub fn prod(&self) -> bool {
-        self.dev.as_ref().map_or(true, DevMode::prod)
-            && self.groups.as_ref().map_or(true, GroupsSpecification::prod)
+        !self.only_groups
     }
 
-    /// Returns `true` if the specification is limited to a select set of groups.
-    pub fn only(&self) -> bool {
-        self.dev.as_ref().is_some_and(DevMode::only)
-            || self.groups.as_ref().is_some_and(GroupsSpecification::only)
+    /// Returns `true` if the specification includes the given group.
+    pub fn contains(&self, group: &GroupName) -> bool {
+        // exclude always trumps include
+        !self.exclude.contains(group) && self.include.contains(group)
     }
 
-    /// Returns the flag that was used to request development dependencies, if specified.
-    pub fn dev_mode(&self) -> Option<&DevMode> {
-        self.dev.as_ref()
+    /// Iterate over all groups that we think should exist.
+    pub fn desugarred_names(&self) -> impl Iterator<Item = &GroupName> {
+        self.include.names().chain(&self.exclude)
     }
 
-    /// Returns the list of groups to include, if specified.
-    pub fn groups(&self) -> Option<&GroupsSpecification> {
-        self.groups.as_ref()
-    }
-
-    /// Returns `true` if the group is included in the specification.
-    pub fn contains(&self, group: &GroupName) -> bool {
-        if group == &*DEV_DEPENDENCIES {
-            match self.dev.as_ref() {
-                None => {}
-                Some(DevMode::Exclude) => {
-                    // If `--no-dev` was provided, always exclude dev.
-                    return false;
-                }
-                Some(DevMode::Only) => {
-                    // If `--only-dev` was provided, always include dev.
-                    return true;
-                }
-                Some(DevMode::Include) => {
-                    // If `--no-group dev` was provided, exclude dev.
-                    return match self.groups.as_ref() {
-                        Some(GroupsSpecification::Include { exclude, .. }) => {
-                            !exclude.contains(group)
-                        }
-                        _ => true,
-                    };
-                }
-            }
-        }
-
-        self.groups
-            .as_ref()
-            .is_some_and(|groups| groups.contains(group))
+    /// Iterate over all groups the user explicitly asked for on the CLI
+    pub fn explicit_names(&self) -> impl Iterator<Item = &GroupName> {
+        let DevGroupsSpecificationHistory {
+            // FIXME: should this being Some add "dev" to the list?
+            dev_mode: _,
```

Comment:
Is this blocking merge? Why wouldn't it add dev to the list?

Reply:
I went back and checked and yeah if we try to complain about `--dev` being passed with no defined dev group it breaks some tests. Updated the comment to reflect this.